worker/transcribee_worker/worker.py

```diff
@@ -0,0 +1,154 @@
+        self,
+        base_url: str,
+        token: str,
+        task_types: Optional[list[TaskType]] = None,
+    ):
+        self.base_url = base_url
+        self.token = token
+        self.tmpdir = None
+        if task_types is not None:
+            self.task_types = task_types
+        else:
+            self.task_types = [TaskType.DIARIZE, TaskType.ALIGN, TaskType.TRANSCRIBE]
+
+    def _get_headers(self):
+        return {"Authorization": f"Worker {self.token}"}
+
+    def claim_task(self) -> Optional[AssignedTask]:
+        logging.info("Asking backend for new task")
+        req = requests.post(
+            f"{self.base_url}/claim_unassigned_task/",
+            params={"task_type": ",".join(self.task_types)},
+            headers=self._get_headers(),
+        )
+        req.raise_for_status()
+        return parse_raw_as(Optional[AssignedTask], req.text)
+
+    def _get_tmpfile(self, filename: str) -> Path:
+        if self.tmpdir is None:
+            raise ValueError("`tmpdir` must be set")
+        return self.tmpdir / filename
+
+    def get_document_file(self, document: ApiDocument) -> Optional[Path]:
+        logging.debug(f"Getting file. {document=}")
+        if document.audio_file is None:
+            return
+        file_url = urllib.parse.urljoin(self.base_url, document.audio_file)
+        outfile = self._get_tmpfile("audiofile")
+        logging.info(f"Downloading {file_url=} to {outfile=}")
+        with open(outfile, "wb") as f:
+            response = requests.get(file_url, stream=True)
+            for data in response.iter_content(chunk_size=1024 * 1024):
+                f.write(data)
+        return outfile
+
+    def keepalive(self, task_id: str, progress: Optional[float]):
+        body = {}
+        if progress is not None:
+            body["progress"] = progress
+        logging.debug(f"Sending keepalive for {task_id=}: {body=}")
+        req = requests.post(
+            f"{self.base_url}/{task_id}/keepalive/",
+            json=body,
+            headers=self._get_headers(),
+        )
+        req.raise_for_status()
+
+    async def perform_task(self, task: AssignedTask):
+        logging.info(f"Running task: {task=}")
+
+        if task.task_type == TaskType.DIARIZE:
+            await self.diarize(task)
+        elif task.task_type == TaskType.TRANSCRIBE:
+            await self.transcribe(task)
+        elif task.task_type == TaskType.ALIGN:
+            await self.align(task)
+        else:
+            raise ValueError(f"Invalid task type: '{task.task_type}'")
+
+    async def transcribe(self, task: TranscribeTask):
+        if task.task_type != TaskType.TRANSCRIBE:
+            return
+
+        document_file = self.get_document_file(task.document)
+        audio = load_audio(str(document_file))
+
+        def progress_callback(_ctx, progress, _data):
+            self.keepalive(task.id, progress=progress / 100)
+
+        paragraphs = []
+        async for paragraph in transcribe(
```

Comment:
This could use async list comprehension

Reply:
I belive it is fine for now as the worker throws a `NotImplemented` anyways currently.